### DISCUSSION TITLE
miluvs中混合搜索查询不到，标量可以，想问问混合搜索中即使不匹配也会返回对应的标量结果还是返回空值？

### DISCUSSION THREAD

**Discussion by javayydd** [milvus-io/milvus#26422] [Q&A and General discussion] [answered]
d单独用标量查询可以有结果：
<img width="1642" alt="image" src="https://github.com/milvus-io/milvus/assets/87173166/bc9d5e50-3972-4e8f-80d0-24d6737bbb50">
单独用矢量查询也有结果，但是不准确：
<img width="1727" alt="image" src="https://github.com/milvus-io/milvus/assets/87173166/c33b993e-2650-4f5d-8f33-56da93f841e2">
使用混合搜索的时候，方法如下：
<img width="1184" alt="image" src="https://github.com/milvus-io/milvus/assets/87173166/7d9c1779-6485-41ff-b991-05e7f14f770a">
查出来是个空值，我看先前有个问题已经回答了，混合搜索
是先检索标量过滤，然后进行矢量的ANN，那么我这里的结果为什么是空，因为标量已经检索了，即使不相似，会返回结果吗？

**Comment by yhmo** [comment, marked as answer]
因为你这里用的索引是IVF，并且设置了nprobe=16，这种索引是把数据分成nlist个桶，搜索的时候仅在nprobe个桶里搜索。
如果符合 "chain_no=="1234567""的数据不在这16个桶里，那么就一条都搜不出来。

打个比方：
一个小学有1000个小学生，有很多班级。如果你用“张三”这个名字去找人，这个学校里如果有叫这个名字的小学生，那么你就找得到。
但现在你仅在三年级的几个班里找，碰巧这个叫张三的学生是五年级的，那你就找不到。

**Reply by javayydd** [reply]
我这里设置的nlist是1024，如果nprobe=16，是不是表明只在16个桶里检索呢，那如果设置nprobe=1024会不会有结果，请给个指导性的建议呗

**Comment by javayydd** [comment]
那我应该怎样设置，才能确保找到数据

**Reply by yhmo** [reply]
把nprobe设置成和nlist相同的值，就能确保找到数据。

**Reply by javayydd** [reply]
谢谢你的回答，我已经找到答案了，请问nprobe是nlist，最高的nprobe个，方便给个指导性的配置这两个参数的意见吗
